Mention new `pub publish` warning in "Publishing packages" (#6247)
dart-lang/site-www: src/content/tools/pub/publishing.md

1. Verifies that your package follows the [pubspec format][pubspec] and
   [package layout conventions][pkg-layout].

> 1. Validates that `git status` is clean (warns against publishing files that
>    are tracked in git, but have uncommitted changes).

Reviewer: This is a long parenthetical. Consider making it part of the sentence or a new second sentence.

Would something like this work?

```suggestion
1. Validates that `git status` is clean. Warns if
   files that are tracked in git have uncommitted changes.
```
